waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -176,21 +176,23 @@ private boolean includeInResults(MetaStoreMapping metaStoreMapping) {
   private boolean includeInResults(MetaStoreMapping metaStoreMapping, String prefixedDatabaseName) {
     return includeInResults(metaStoreMapping)
         && isWhitelisted(metaStoreMapping.getDatabasePrefix(),
-            metaStoreMapping.transformInboundDatabaseName(prefixedDatabaseName));
+        metaStoreMapping.transformInboundDatabaseName(prefixedDatabaseName));
   }
 
   @Override
   public DatabaseMapping databaseMapping(@NotNull String databaseName) {
     // Find a Metastore with a prefix
-    Iterator<Entry<String, DatabaseMapping>> iterator = mappingsByPrefix.entrySet().iterator();
-    while (iterator.hasNext()) {
-      Entry<String, DatabaseMapping> entry = iterator.next();
-      String metastorePrefix = entry.getKey();
-      if (Strings.isNotBlank(metastorePrefix) && databaseName.startsWith(metastorePrefix)) {
-        DatabaseMapping databaseMapping = entry.getValue();
-        LOG.debug("Database Name `{}` maps to metastore with prefix `{}`", databaseName, metastorePrefix);
-        if (includeInResults(databaseMapping, databaseName)) {
-          return databaseMapping;
+    synchronized (mappingsByPrefix) {
```

Comment:
1. Why are we synchronizing? 2. and why on `mappingsByPrefix`?

Reply:
The map is synchroniszed, we're looping over it, javadoc states the pattern: https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedMap-java.util.Map-

This synchronisation wasn't done in a couple of places which I think is a bug on our part, so we rectified that.

Reply:
OK